evidence_test.go

```diff
@@ -43,166 +44,72 @@ func mustBuildValidCcaPlatformClaims(t *testing.T, includeOptional bool) psatoke
+	err := EvidenceIn.SetCcaPlatformClaims(
+		mustBuildValidCcaPlatformClaims(t, true),
+	)
 	assert.NoError(t, err)
 
-	cwt, err := EvidenceIn.Sign(PlatformSigner, RealmSigner)
-	assert.NoError(t, err, "signing failed")
-
-	fmt.Printf("CCA evidence : %x\n", cwt)
-
-	var EvidenceOut CcaEvidence
-
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
-
-	pkPlatform := pubKeyFromJWK(t, testECKeyA)
-	pkRealm := pubKeyFromJWK(t, testECKeyA)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
+	err = EvidenceIn.SetCcaRealmClaims(
+		mustBuildValidCcaRealmClaims(t),
+	)
 	assert.NoError(t, err)
-}
 
-func TestEvidence_sign_and_verify_non_identical_key_ok(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyB)
-	RealmSigner := signerFromJWK(t, testECKeyA)
-	var EvidenceIn CcaEvidence
-
-	err := EvidenceIn.SetCcaPlatformClaims(mustBuildValidCcaPlatformClaims(t, true))
-	assert.NoError(t, err)
-
-	err = EvidenceIn.SetCcaRealmClaims(mustBuildValidCcaRealmClaims(t))
-	assert.NoError(t, err)
-
-	cwt, err := EvidenceIn.Sign(PlatformSigner, RealmSigner)
+	ccaToken, err := EvidenceIn.Sign(pSigner, rSigner)
 	assert.NoError(t, err, "signing failed")
 
-	fmt.Printf("CCA evidence : %x\n", cwt)
+	fmt.Printf("CCA evidence : %x\n", ccaToken)
 
-	var EvidenceOut CcaEvidence
+	var EvidenceOut Evidence
 
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
+	err = EvidenceOut.FromCBOR(ccaToken)
+	assert.NoError(t, err, "CCA token decoding failed")
 
-	pkPlatform := pubKeyFromJWK(t, testECKeyB)
-	pkRealm := pubKeyFromJWK(t, testECKeyA)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
-	assert.NoError(t, err)
-}
-
-func TestEvidence_sign_and_verify_realm_key_mismatch(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyB)
-	RealmSigner := signerFromJWK(t, testECKeyA)
-	var EvidenceIn CcaEvidence
-
-	err := EvidenceIn.SetCcaPlatformClaims(mustBuildValidCcaPlatformClaims(t, true))
-	assert.NoError(t, err)
+	verifier := pubKeyFromJWK(t, testIAK)
 
-	err = EvidenceIn.SetCcaRealmClaims(mustBuildValidCcaRealmClaims(t))
+	err = EvidenceOut.Verify(verifier)
 	assert.NoError(t, err)
-
-	cwt, err := EvidenceIn.Sign(PlatformSigner, RealmSigner)
-	assert.NoError(t, err, "signing failed")
-
-	fmt.Printf("CCA evidence : %x\n", cwt)
-
-	var EvidenceOut CcaEvidence
-
-	err = EvidenceOut.FromCBOR(cwt)
-	assert.NoError(t, err, "evidence contaiting CCA token decoding failed")
-
-	pkPlatform := pubKeyFromJWK(t, testECKeyB)
-	pkRealm := pubKeyFromJWK(t, testECKeyB)
-	err = EvidenceOut.Verify(pkPlatform, pkRealm)
-	assert.EqualError(t, err, "unable to verify realm token: verification error")
 }
 
 func TestEvidence_sign_and_verify_platform_key_mismatch(t *testing.T) {
-	PlatformSigner := signerFromJWK(t, testECKeyA)
-	RealmSigner := signerFromJWK(t, testECKeyA)
-	var EvidenceIn CcaEvidence
+	rSigner := signerFromJWK(t, testRAK)
```

Comment:
I think, we should have a similar test for verification where the Key used for signing the Realm token does not correspond to the Realm Public Key in the token

Reply:
sgtm